src/guides/v2.3/extension-dev-guide/price-adjustments.md

```diff
@@ -0,0 +1,411 @@
+---
+group: php-developer-guide
+title: Price Adjustments
+contributor_name: Goivvy LLC
+contributor_link: https://www.goivvy.com
+---
+
+## How Price Adjustments Work
+
+Price Adjustments will adjust product price as it's displayed on category or product pages.  
+
+**In this article, as an example, we will add `1.79` to each product price.**
+
+To introduce new price adjustment one should add the following code to module's `etc/di.xml`:
+
+```xml
+<type name="Magento\Framework\Pricing\Adjustment\Collection">
+  <arguments>
+    <argument name="adjustments" xsi:type="array">
+      <item name="devadj" xsi:type="const">VENDOR\MODULE\Pricing\Adjustment::ADJUSTMENT_CODE</item>
+    </argument>
+  </arguments>
+</type>
+<type name="Magento\Framework\Pricing\Adjustment\Pool">
+  <arguments>
+     <argument name="adjustments" xsi:type="array">
+        <item name="devadj" xsi:type="array">
+           <item name="className" xsi:type="string">VENDOR\MODULE\Pricing\Adjustment</item>
+           <item name="sortOrder" xsi:type="string">10</item>
+        </item>
+     </argument>
+  </arguments>
+</type>
+```
+
+Then `VENDOR/MODULE/Pricing/Adjustment.php` file should look like this:
+
+```php
+namespace VENDOR\MODULE\Pricing;
+
+use Magento\Framework\Pricing\Adjustment\AdjustmentInterface;
+use Magento\Framework\Pricing\SaleableInterface;
+
+class Adjustment implements AdjustmentInterface
+{
+
+    const ADJUSTMENT_CODE = 'devadj';
+    const ADJUSTMENT_VALUE = 1.79;
+
+    /**
+     * Get adjustment code
+     *
+     * @return string
+     */
+    public function getAdjustmentCode()
+    {
+        return self::ADJUSTMENT_CODE;
+    }
+
+    /**
+     * Define if adjustment is included in base price
+     *
+     * @return bool
+     */
+    public function isIncludedInBasePrice()
+    {
+        return true;
+    }
+
+    /**
+     * Define if adjustment is included in display price
+     *
+     * @return bool
+     */
+    public function isIncludedInDisplayPrice()
+    {
+        return true;
+    }
+
+    /**
+     * Extract adjustment amount from the given amount value
+     *
+     * @param float $amount
+     * @param SaleableInterface $saleableItem
+     * @param null|array $context
+     * @return float
+     */
+    public function extractAdjustment($amount, SaleableInterface $saleableItem, $context = [])
+    {
+        $return = $amount - self::ADJUSTMENT_VALUE;
```

Comment:
I would not recommend declaring the redundant `$return` variable if we don't need storing the data. Returning a result is enough in this case and the following cases.

```suggestion
        return $amount - self::ADJUSTMENT_VALUE;
```

From the interpreter perspective, the system needs to allocate separate memory space for the variable, then a separate space for the result of the function, then free up the space for the variable. We don't need half of the operations under the hood from the performance perspective.

It's not a big deal since it's just an example. But my intention is to show the best practices in the DevDocs examples. 

Thank you